lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -845,6 +875,30 @@ export abstract class ClientApplication {
         }
     }
 
+    /**
+     * Returns the signed in account matching the tenant profile passed in or null if no account is found.
+     * @param tenantProfile
+     * @returns The account object stored in MSAL 
+     */
+    getAccountByTenantProfile(tenantProfile: TenantProfile): AccountInfo | null {
```

Comment:
Rather than adding another getAccount API can we add a `getAccountByFilter` API so we don't need to keep adding additional APIs in the future? Let's also mark the existing getAccountBy APIs deprecated and point folks to the new one

Reply:
Done. I did ad a new account API called `getAccountsFilteredBy` that returns all the accounts that match the filter, not just the first. It enables a getAllAccounts with filter "isHomeTenant" to only return home or guest accounts depending on the app needs.